infra/scripts/release-patch.sh

```diff
@@ -0,0 +1,279 @@
+    -t, --github-token      personal GitHub token
+    -r, --remote            git remote server name for the feast-dev/feast repo (e.g. origin, upstream, etc.)
+"
+}
+
+while [ "$1" != "" ]; do
+  case "$1" in
+      -v | --version )           VERSION="$2";            shift;;
+      -t | --github-token )      GH_TOKEN="$2";           shift;;
+      -r | --remote )            REMOTE="$2";             shift;;
+      -h | --help )              usage;                   exit;;
+      * )                        usage;                   exit 1
+  esac
+  shift
+done
+
+if [ -z $VERSION ]; then usage; exit 1; fi
+if [ -z $GH_TOKEN ]; then usage; exit 1; fi
+if [ -z $REMOTE ]; then usage; exit 1; fi
+regex="([0-9]+)\.([0-9]+)\.([0-9]+)"
+if [[ $VERSION =~ $regex ]]
+then
+    MAJOR="${BASH_REMATCH[1]}"
+    MINOR="${BASH_REMATCH[2]}"
+    PATCH="${BASH_REMATCH[3]}"
+else
+    usage
+    exit 1
+fi
+if ! which gh ; then echo "Please install the GitHub CLI to use this script"; exit 1; fi
+
+echo "This script is mostly idempotent; check git status for temp files before restarting. It will always prompt you before making any non-local change."
+
+# Go to infra/scripts directory
+cd $(dirname "$0")
+# Login to GitHub CLI
+echo $GH_TOKEN | gh auth login --with-token
+
+echo "Step 1: rebase new commits onto release branch"
+git fetch $REMOTE
+git checkout $REMOTE/master
+STARTING_COMMIT=$(git merge-base $REMOTE/master v$MAJOR.$MINOR-branch)
+git checkout v$MAJOR.$MINOR-branch
+
+push_rebased_commits()
+{
+    echo "Pushing commits"
+    git push $REMOTE v$MAJOR.$MINOR-branch
+    echo "Commits pushed"
+}
+rebase_from_master()
+{
+    echo "Rebasing commits"
+    git checkout $REMOTE/master
+    git rebase --interactive --onto v$MAJOR.$MINOR-branch $STARTING_COMMIT HEAD
+    git branch -f v$MAJOR.$MINOR-branch HEAD
+    git checkout v$MAJOR.$MINOR-branch
+    echo "Commits rebased"
+    echo "Step 1b: Push commits"
+    read -p "Commits are not pushed. Continue (y) or skip this sub-step (n)? " choice
+    case "$choice" in
+        y|Y ) push_rebased_commits ;;
+        n|N ) echo "Skipping this sub-step" ;;
+        * ) echo "Skipping this sub-step" ;;
+    esac ;
+}
+echo "Step 1a: rebase commits"
+if git status | grep -q "is ahead of" ; then
+    read -p "Your local branch is ahead of its remote counterpart, indicating you may have already rebased. Skip this step (y) or run the rebase starting from commit $STARTING_COMMIT (n)? " choice
+    case "$choice" in
+        y|Y ) echo "Skipping this step" ;;
+        n|N ) rebase_from_master ;;
+        * ) rebase_from_master ;;
+    esac ;
+else
+    read -p "Will rebase starting from commit $STARTING_COMMIT. Continue (y) or skip this step (n)? " choice
+    case "$choice" in
+        y|Y ) rebase_from_master ;;
+        n|N ) echo "Skipping this step" ;;
+        * ) echo "Skipping this step" ;;
+    esac ;
+fi
+
+commit_changelog()
+{
+    echo "Committing CHANGELOG.md"
+    git add ../../CHANGELOG.md
```

Comment:
Can we use `git rev-parse --show-toplevel` to get the root of the git repo, so that this `git add` doesn't have to be run from a specific directory?